incubator/gocd/templates/gocd-server-deployment.yaml

```diff
@@ -0,0 +1,75 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "gocd.fullname" . }}-server
+  labels:
+    app: {{ template "gocd.name" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
+    component: server
+spec:
+  {{- if le (int .Values.server.replicaCount) 1 }}
+  replicas: {{ .Values.server.replicaCount }}
```

Comment:
Why do you make this configurable at all? This doesn't seem to make sense.

Reply:
Supported values for GoCD Server replicas Count are `0` and `1`. Where,
`0` indicates `Agent Only Deployment` and
`1` indicates `Agent-Server Deployment`.

One can choose to deploy only GoCD agents on the kubernetes cluster. In such cases the `server.replicaCount` value will be `0`.

Reply:
Is `0` a valid value? Didn't know that but seems strange to me. Why don't you add an if statement around the deployment? That would be the usual way how this is done. And make it a boolean value.

Reply:
I've changed the `server.replicaCount` variable to `server.enabled` variable which can have boolean value
`true`: `agent-server deployment`  and
`false`: `agent-only deployment`.